src/Main.js

```diff
@@ -4,30 +4,31 @@ import ExtendedView from './ExtendedView'
 import List from './List'
 import movieData from './mockData'
 import './Main.css'
+import {getAllMovies, fetchSingleMoviePoster} from './apiCalls'
 
 class Main extends React.Component {
   constructor() {
     super();
     this.state = {
       movieData: [],
+      singleMovie: [],
       isExtendedView: false,
       clickedPosterID: null,
-      loading: false
+      loading: false,
+      error: ''
     }
   }
 
   componentDidMount() {
     console.log("did mount")
     this.setState({loading: true})
-    // const url = 'https://rancid-tomatillos.herokuapp.com/api/v2/movies'
-      fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
-       .then(response => response.json())
-       .then(data => {
-         this.setState({
-           loading: false,
-           movieData: data.movies
-         })
-       })
+    getAllMovies()
+    .then(data => {
+      this.setState({
+        loading: false,
+        movieData: data.movies
+      })
+    })
        .catch(error => console.log("error"))
```

Comment:
We will need to start doing some more robust error handling soon.

We could use a method that would render an Error component based off of a property value in state.

Reply:
Yes!